[Dynamo][Guard]Add the user-friendly TYPE_MATCH for type

fxdawnn opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #169025

Fix #168160 after the opensource PR https://github.com/pytorch/pytorch/pull/168272

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela @jataylo

fxdawnn committed:
[Dynamo][Guard]Add the user-friendly TYPE_MATCH for type

[ghstack-poisoned]

anijain2305 reviewed (commented):
LGTM. Just minor test change requests.

fxdawnn committed:
Update on "[Dynamo][Guard]Add the user-friendly TYPE_MATCH for type"


Fix #168160 after the opensource PR https://github.com/pytorch/pytorch/pull/168272

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

anijain2305 reviewed (approved):
LGTM!

fxdawnn committed:
Update on "[Dynamo][Guard]Add the user-friendly TYPE_MATCH for type"


Fix #168160 after the opensource PR https://github.com/pytorch/pytorch/pull/168272

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

fxdawnn committed:
Update on "[Dynamo][Guard]Add the user-friendly TYPE_MATCH for type"


Fix #168160 after the opensource PR https://github.com/pytorch/pytorch/pull/168272

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

fxdawnn committed:
Update on "[Dynamo][Guard]Add the user-friendly TYPE_MATCH for type"


Fix #168160 after the opensource PR https://github.com/pytorch/pytorch/pull/168272

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

fxdawnn committed:
Update on "[Dynamo][Guard]Add the user-friendly TYPE_MATCH for type"


Fix #168160 after the opensource PR https://github.com/pytorch/pytorch/pull/168272

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

fxdawnn committed:
Update on "[Dynamo][Guard]Add the user-friendly TYPE_MATCH for type"


Fix #168160 after the opensource PR https://github.com/pytorch/pytorch/pull/168272

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

fxdawnn committed:
Update on "[Dynamo][Guard]Add the user-friendly TYPE_MATCH for type"


Fix #168160 after the opensource PR https://github.com/pytorch/pytorch/pull/168272

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

fxdawnn commented:
@pytorchbot merge

fxdawnn committed:
Update

[ghstack-poisoned]

fxdawnn commented:
@pytorchbot merge -r

fxdawnn committed:
Update

[ghstack-poisoned]

fxdawnn committed:
Update on "[Dynamo][Guard]Add the user-friendly TYPE_MATCH for type"


Fix #168160 after the opensource PR https://github.com/pytorch/pytorch/pull/168272

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

fxdawnn commented:
@pytorchbot merge

fxdawnn committed:
Update on "[Dynamo][Guard]Add the user-friendly TYPE_MATCH for type"


Fix #168160 after the opensource PR https://github.com/pytorch/pytorch/pull/168272

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

fxdawnn commented:
@pytorchbot merge

fxdawnn committed:
Update on "[Dynamo][Guard]Add the user-friendly TYPE_MATCH for type"


Fix #168160 after the opensource PR https://github.com/pytorch/pytorch/pull/168272

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

huydhn commented:
@pytorchbot revert -m 'Abandoning D88461499 seems to lead to a discrepancy between github and the import diff.  I need to revert and reland this to reimport the change.  If your change only touch GitHub OSS, you dont need to do anything with the import diff on diff train and can just reland it' -c ghfirst

huydhn commented:
@pytorchbot merge -f 'Reland this back'
